log/field.go

```diff
@@ -0,0 +1,204 @@
+package log
```

Comment:
I think the `LazyLogger` will be sufficient for things like `Reference` logging (which we may not even special-case), so I moved this back into its own package.